tools/functions.sh

```diff
@@ -137,9 +139,9 @@ validate_file() {
     local checksums_path="${1}"
     local file_path="${2}"
     local cmd="${3}"
-    if ! grep $(openssl dgst -sha256 ${file_path} | cut -d ' ' -f 2) ${checksums_path}; then
-        echo ${cmd} is not a valid file
-        rm -rf ${file_path}
+    if ! grep "$(openssl dgst -sha256 "${file_path}" | cut -d ' ' -f 2)" "${checksums_path}"; then
```

Comment:
I'm not sure what is the intended behavior here, but the I think either the `"${file_path}"` should be `\"${file_path}\"` or `'${file_path}'`.

Or can we just move out the command into its own line? Like:
```bash
local digest=$(openssl dgst -sha256 "${file_path}" | cut -d ' ' -f 2)
if ! grep "${digest}" "${checksums_path}"; then
....
```

Reply:
yeh i think i prefer that idea